docs/konnectors_oauth.md

```diff
@@ -0,0 +1,149 @@
+# Doctypes
+
+`io.cozy.konnectors` gives their desiderata for an account
+
+```json
+{
+  "fields": {
+    "account": {
+      "doctype": "io.cozy.accounts",
+      "account_type": "trainlines",
+      "accountFormat": "login,password",
+    }
+  },
+
+}
+```
+
+`io.cozy.accounts` contains authentification information for an account
+
+```json
+{
+    "name" : "Gmail Perso",
+    "account_type": "google",
+    "status": "connected",
+    "auth": {
+      "user": "my-personal-account@gmail.com",
+      "password": "my-secret"
+    }
+}
+```
+
+`io.cozy.accounts_type` contains the oauthconfiguration
+
+```json
+{
+  "_id":"google",
+  "mode": "userpass"
+}
+```
+
+```json
+{
+  "_id":"service.example",
+  "mode": "oauth-server-flow",
+  "client_id": "",
+  "client_secret": "",
+  "auth_endpoint": "https://service.example/auth",
+  "token_endpoint":  "https://api.service.example/token"
+}
+```
+
+@TODO some encryption of cozy client secret might be useful
+
+# Reminder OAuth flows
+
+**Service** is the website konnector aims to retrieve informations from
+
+## webserver flow (Authorization Code Grant)
+
+A. In SettingsApp give a link
+  <a href="https://service.example/auth? (url)
+      response_type=code&
+      client_id=CLIENT_ID&
+      redirect_uri=https://bob.cozy.rocks/accounts/redirect&
+      scope=photos&
+      state=1234zyx"
+
+@NOTE the scope may depends on other fields being configured (checkboxes)
+
+B. Service let the user login, allow or deny scope
+Then redirect to https://cozy.rocks/accounts/redirect? (url)
+  code=AUTH_CODE_HERE&
+  state=1234zyx
+
+C. The Stack does Server side
+POST https://api.service.example/token (form)
+  grant_type=authorization_code&
+  code=AUTH_CODE_HERE&
+  redirect_uri=https://bob.cozy.rocks/accounts/redirect&
+  client_id=CLIENT_ID&
+  client_secret=CLIENT_SECRET
+
+D. The Service responds (serverside) with (json)
+{
+  "access_token":"ACCESS_TOKEN",
+  "token_type":"bearer",
+  "expires_in":2592000,
+  "refresh_token":"REFRESH_TOKEN",
+  "scope":"read",
+  "uid":100101,
+  "info":{"name":"Claude Douillet","email":"claude.douillet@example.com"}
+}
+This whole object is saved as-is into a `io.cozy.accounts`
+
+@TODO better to save known fields separately from other random infos.
+
+
+E. The Stack redirect the user to SettingsApp with configured account.
+
+@TODO state should contains enough information for settings app to come back to previous state (just before user clicked the link)
```

Comment:
Should we advise the settings app to store its state in the local storage, and use a random number for the `state` parameter of OAuth (this number is also the key of the state in the local storage)?